core/org.openjdk.jmc.flightrecorder.serializers/src/main/java/org/openjdk/jmc/flightrecorder/serializers/json/IItemCollectionJsonSerializer.java

```diff
@@ -59,7 +59,7 @@
 public class IItemCollectionJsonSerializer extends JsonWriter {
 	private final static Logger LOGGER = Logger.getLogger("org.openjdk.jmc.flightrecorder.json");
 
-	public static String toJsonString(IItemCollection items) {
+	public synchronized static String toJsonString(IItemCollection items) {
```

Comment:
I'm not sure putting `synchronized` on this method in core libraries is the right call. Every consumer of this method will now pay the price for a JMC issue.

I think it's better to fix the issue where the web based views use this methods.

----

Also on the topic of synchronization I believe it's better to synchronize in the method body rather than the method.